Получить все рекомендации пользователя

s3nkyh opened the pull request.

Commit:
added: RecomContrl, RecomDto, RecomService, SkillOfferDto

Commit:
--

Commit:
-

Commit:
--

Commit:
исправил RecommendationMapper

Commit:
убрал лишние аннотации

Commit:
исправил RecommendationMapper
добавил метод deleteAllAndCreate в RecommendationService

Commit:
реализовал механику удаления рекомендации

Commit:
реализовал механику получение всех данных рекомендации

Commit:
реализовал механику получения всех данных рекомендации

Commit:
реализовал механику получения всех рекомендаций пользователя

Commit:
исправил ошибку в проверку на null

SergeyErmoshenko reviewed (changes requested):
С учетом замечаний в прошлом PR

Commit:
исправил RecommendationService,
RecommendationValidation,
RecommendationMapper
добавил метод в RecommendationRepository и SkillOfferRepository

SergeyErmoshenko commented on src/main/java/school/faang/user_service/service/RecommendationService.java:
```diff
@@ -0,0 +1,102 @@
+        Page<Recommendation> recommendationPage = recommendationRepository.findAllByAuthorId(authorId, pageable);
+        List<Recommendation> recommendationList = recommendationPage.getContent();
+        return recommendationList.stream()
+                .map(recommendationMapper::toRecommendationDto)
```
Можно сразу использовать мап листа в toRecommendationDto. И ниже тоже исправь

Commit:
исправил RecommendationService,
RecommendationMapper
